app/controllers/internal/events_controller.rb

```diff
@@ -44,6 +44,11 @@ def create
     def edit
       api_event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:id])
       @event = Event.initialize_with_api_event(api_event)
+      if params[:duplicated]
```

Comment:
I think this is a bit counter-intuitive having it in the `edit` action when you're not editing, you're creating - I would have expected this in the `new` action, something like:

```
/events/new?duplicate=<event_id>
```

Reply:
@ethax-ross Updated based on this ^